src/bio/bio_buffer.c

```diff
@@ -886,10 +897,12 @@ nvme_rw(struct bio_desc *biod, struct bio_rsrvd_region *rg)
 	xs_ctxt->bxc_blob_rw++;
 
 	D_DEBUG(DB_IO, "%s blob:%p payload:%p, pg_idx:"DF_U64", "
-		"pg_cnt:"DF_U64"\n", biod->bd_update ? "Write" : "Read",
+		"pg_cnt:"DF_U64"\n",
+		biod->bd_type == BIO_IOD_TYPE_UPDATE ? "Write" : "Read",
 		blob, payload, pg_idx, pg_cnt);
 
-	if (biod->bd_update)
+	D_ASSERT(biod->bd_type < BIO_IOD_TYPE_GETBUF);
```

Comment:
can this be checked sooner (already checked in dma_rw() right?).  is there a case where this will be called if not a read/write? or just extra protection?

Reply:
Right, it's just extra protection. In case of someone else changing the code for some new features. :)